src/crypto.ts

```diff
@@ -195,4 +200,41 @@ export default class Crypto {
       decodeBase64(formSecretKey)
     )
   }
+
+  /**
+   * Download and decrypt the given attachments.
+   * @param formSecretKey Secret key as a base-64 string
+   * @param decryptParams The params containing encrypted content and information.
+   * @returns The decrypted attachments if successful. Else, null will be returned.
+   * @throws {MissingPublicKeyError} if a public key is not provided when instantiating this class and is needed for verifying signed content.
+   */
+  downloadAndDecryptAttachments = async(formSecretKey: string, decryptParams: DecryptParams): Promise<DecryptedAttachments | null> => {
+    let decryptedRecords: DecryptedAttachments = {}
+
+    const attachmentRecords: EncryptedAttachmentRecords = decryptParams.attachmentDownloadUrls ?? {}
+    const decryptedContent = this.decrypt(formSecretKey, decryptParams)
+    if (decryptedContent === null) return null
+
+    for (let fieldId in attachmentRecords) {
+      const downloadResponse = await axios.get(attachmentRecords[fieldId], { responseType: 'json' })
+      if (downloadResponse.status !== 200) return null
+
+      let data = downloadResponse.data
+      data.encryptedFile.binary = decodeBase64ToUint8Array(data.encryptedFile.binary)
+      const decryptedFile = await this.decryptFile(formSecretKey, data.encryptedFile)
+
+      let filename = 'Unknown File.data'
```

Comment:
can we afford to be more strict here, eg throw an error if we find an attachment with an ID that does not actually exist in the responses? might be strange UX if users see "Unknown File.data" in their downloads, especially since we aren't able to offer any scanning for malicious files

Reply:
I think in this case we should return null because there is obviously something wrong with the attachment.